internal/server/routes.go

```diff
@@ -123,6 +123,14 @@ func (s *Server) GenerateRoutes() Routes {
 	get(a, noAuthnWithOrg, "/api/providers/:id", a.GetProvider)
 	get(a, noAuthnWithOrg, "/api/providers", a.ListProviders)
 	get(a, noAuthnWithOrg, "/api/settings", a.GetSettings)
+	add(a, noAuthnWithOrg, route[api.VerifyAndRedirectRequest, *api.RedirectResponse]{
+		method:                     http.MethodGet,
+		path:                       "/link",
```

Comment:
This should probably be `/api/link`, so that we keep all paths under `/api`. Otherwise problems like #3161 become a lot harder to fix.

Thoughts on `/api/verify` instead of `/api/link` ? I guess this is trying to obscure what it's doing? Or is the idea that we might use this for other purposes later?

Reply:
it's not strictly related to the API. If I put api in the link it'll be visible in links in emails, which I don't think is ideal.
It's not important to the user that it's for verification, and if the user has already verified, seeing verification in the link could be confusing; the link will still function properly.